projects/components/src/gauge/gauge.component.ts

```diff
@@ -0,0 +1,175 @@
+import { ChangeDetectionStrategy, Component, ElementRef, Input, OnChanges } from '@angular/core';
+import { Color, LayoutChangeService, Point } from '@hypertrace/common';
+import { Arc, arc, DefaultArcObject } from 'd3-shape';
+import { BehaviorSubject, combineLatest, Observable, Subject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'ht-gauge',
+  template: `
+    <svg #chartContainer class="gauge" (htLayoutChange)="this.onLayoutChange()" *ngIf="this.gaugeRendererData$ | async as rendererData">
+      <g attr.transform="translate({{ rendererData.origin.x }}, {{ rendererData.origin.y }})">
+        <path class="gauge-ring" [attr.d]="rendererData.backgroundArc" />
+        <g
+          class="input-data"
+          *ngIf="rendererData.data"
+          htTooltip="{{ rendererData.data.value }} of {{ rendererData.data.maxValue }}"
+        >
+          <path
+            class="value-ring"
+            [attr.d]="rendererData.data.valueArc"
+            [attr.fill]="rendererData.data.threshold.color"
+          />
+          <text x="0" y="0" class="value-display" [attr.fill]="rendererData.data.threshold.color">
+            {{ rendererData.data.value }}
+          </text>
+          <text x="0" y="24" class="label-display">{{ rendererData.data.threshold.label }}</text>
+        </g>
+      </g>
+    </svg>
+  `,
+  styleUrls: ['./gauge.component.scss'],
+  providers: [LayoutChangeService],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class GaugeComponent implements OnChanges {
+  private static readonly GAUGE_RING_WIDTH: number = 20;
+  private static readonly GAUGE_ARC_CORNER_RADIUS: number = 10;
+  private static readonly GAUGE_AXIS_PADDING: number = 30;
+
+  @Input()
+  public value?: number;
+
+  @Input()
+  public maxValue?: number;
+
+  @Input()
+  public thresholds: GaugeThreshold[] = [];
+
+  public readonly gaugeRendererData$: Observable<GaugeSvgRendererData>;
+
+  private readonly inputDataSubject: Subject<GaugeInputData | undefined> = new BehaviorSubject<
+    GaugeInputData | undefined
+  >(undefined);
+  private readonly inputData$: Observable<GaugeInputData | undefined> = this.inputDataSubject.asObservable();
+
+  private readonly redrawSubject: Subject<true> = new BehaviorSubject(true);
+
+  public constructor(public readonly elementRef: ElementRef) {
+    this.gaugeRendererData$ = this.buildGaugeRendererDataObservable();
+  }
+
+  public ngOnChanges(): void {
+    this.emitInputData();
+  }
+
+  public onLayoutChange(): void {
+    this.redrawSubject.next(true);
```

Comment:
Not wrong, but a bit strange to do this with a subject. Could just as easily remove the subject + combineLatest and do
`this.gaugeRendererData$ = this.buildGaugeRendererDataObservable();`